code/game/turfs/exterior/_exterior.dm

```diff
@@ -0,0 +1,186 @@
+		set_light(owner.lightlevel, 0.1, 2)
+		if(owner.planetary_area && istype(loc, world.area))
+			ChangeArea(src, owner.planetary_area)
+	update_icon(TRUE)
+	. = ..()
+
+/turf/exterior/attackby(obj/item/C, mob/user)
+
+	if(diggable && istype(C,/obj/item/shovel))
+		visible_message(SPAN_NOTICE("\The [user] starts digging at \the [src]."))
+		if(do_after(user, 50))
+			to_chat(user, SPAN_NOTICE("You dig a deep pit."))
+			new /obj/structure/pit(src)
+			diggable = 0
+		else
+			to_chat(user, SPAN_NOTICE("You stop shoveling."))
+		return TRUE
+
+	if(istype(C, /obj/item/stack/tile))
+		var/obj/item/stack/tile/T = C
+		if(T.use(1))
+			playsound(src, 'sound/items/Deconstruct.ogg', 80, 1)
+			ChangeTurf(/turf/simulated/floor, FALSE, FALSE, TRUE)
+		return TRUE
+
+	. = ..()
+
+/turf/exterior/explosion_act(severity)
+	SHOULD_CALL_PARENT(FALSE)
+	if(!istype(src, get_base_turf_by_area(src)) && (severity == 1 || (severity == 2 && prob(40))))
+		ChangeTurf(get_base_turf_by_area(src))
+
+var/exterior_burned_states = icon_states('icons/turf/exterior/burned.dmi')
+/turf/exterior/proc/set_burned(var/skip_update)
+	burned = pick(global.exterior_burned_states)
+	if(!skip_update)
+		queue_icon_update()
+
+var/exterior_broken_states = icon_states('icons/turf/exterior/broken.dmi')
+/turf/exterior/proc/set_broken(var/skip_update)
+	burned = pick(global.exterior_broken_states)
+	if(!skip_update)
+		queue_icon_update()
+
+/turf/exterior/on_update_icon(var/update_neighbors)
+	..() // Recalc AO and flooding overlay.
+	cut_overlays()
+	if(LAZYLEN(decals))
+		add_overlay(decals)
+
+	if(broken)
+		add_overlay(get_damage_overlay(broken, BLEND_MULTIPLY, 'icons/turf/exterior/broken.dmi'))
+	if(burned)
+		add_overlay(get_damage_overlay(burned, BLEND_MULTIPLY, 'icons/turf/exterior/burned.dmi'))
+
+	if(icon_edge_layer < 0)
+		return
+
+	var/neighbors = 0
+	for(var/direction in GLOB.cardinal)
+		var/turf/exterior/turf_to_check = get_step(src,direction)
+		if(!turf_to_check || turf_to_check.density)
+			continue
+		if(istype(turf_to_check, type))
+			neighbors |= direction
+			continue
+		if(!istype(turf_to_check) || icon_edge_layer > turf_to_check.icon_edge_layer)
+			var/image/I = image(icon, "edge[direction][icon_edge_states > 0 ? rand(0, icon_edge_states) : ""]")
+			I.layer = layer + icon_edge_layer
+			switch(direction)
+				if(NORTH)
+					I.pixel_y += world.icon_size
+				if(SOUTH) 
+					I.pixel_y -= world.icon_size
+				if(EAST)
+					I.pixel_x += world.icon_size
+				if(WEST)
+					I.pixel_x -= world.icon_size
+			add_overlay(I)
+		if(update_neighbors)
+			turf_to_check.update_icon()
+
+	if(icon_has_corners || update_neighbors)
+		for(var/direction in GLOB.cornerdirs)
+			var/turf/exterior/turf_to_check = get_step(src,direction)
+			if(!turf_to_check || turf_to_check.density || istype(turf_to_check, type))
+				continue
+
+			var/draw_state
+			switch(direction)
+				if(NORTHEAST)
+					if(!(neighbors & NORTH) && !(neighbors & EAST))
```

Comment:
can't this be simplified as `(neighbors & NORTHEAST) != NORTHEAST`?

Reply:
I don't know.

Reply:
Turns out it can.